packages/flutter_tools/test/integration.shard/test_data/gen_l10n_project.dart

```diff
@@ -246,4 +247,88 @@ void main() {
   }
 }
 ''';
+
+
+}
+
+class GenL10nLocaleResolutionProject extends GenL10nProject {
+  @override
+  Future<void> setUpIn(Directory dir) {
+    this.dir = dir;
+    writeFile(globals.fs.path.join(dir.path, 'lib', 'l10n', 'app_en_CA.arb'), appEnCa);
+    return super.setUpIn(dir);
+  }
+
+  @override
+  String get main => _main;
+  static const String _main = r'''
+import 'package:flutter/material.dart';
+
+import 'l10n/app_localizations.dart';
+
+class Home extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    try {
+      print('get localizations');
+      final AppLocalizations localizations = AppLocalizations.of(context);
+      print('localizations: $localizations');
+      print('localizations.helloWorld');
+      final List<String> results = <String>[
+        '${localizations.helloWorld}',
+        '${localizations.helloCost("price", 123)}',
+        '${localizations.hello("World")}',
+        '${localizations.greeting("Hello", "World")}',
+        '${localizations.helloWorldOn(DateTime(1960))}',
+        '${localizations.helloOn("world argument", DateTime(1960), DateTime(1960))}',
+        '${localizations.helloWorldDuring(DateTime(1960), DateTime(2020))}',
+        '${localizations.helloFor(123)}',
+      ];
+      int n = 0;
+      for (final String result in results) {
+        print('#l10n $n ($result)\n');
+        n += 1;
+      }
+    } finally {
+      print('#l10n END\n');
+    }
+    return Container();
+  }
+}
+
+void main() {
+  runApp(
+    MaterialApp(
+      localizationsDelegates: AppLocalizations.localizationsDelegates,
+      supportedLocales: <Locale>[
+        Locale('en', 'CA'),
+      ],
+      home: Home(),
+    ),
+  );
+}
+''';
+
+  final String appEnCa = r'''
+{
+  "@@locale": "en_CA",
```

Comment:
For now, en_CA.arb will need to contain all of the same messages as en.arb or just the plurals?

Reply:
Just the plurals. I'll add them here with a comment, and we can remove 1-2 for testing once we have implemented our own message lookup